psl-cli/src/test/java/org/linqs/psl/cli/CLITest.java

```diff
@@ -95,6 +107,156 @@ public void run(String modelPath, String dataPath, String loggingLevel, List<Str
         validate();
     }
 
+    public String runOnline(String modelPath, String dataPath, String actionPath) {
+        return runOnline(modelPath, dataPath, actionPath, null);
+    }
+
+    public String runOnline(String modelPath, String dataPath, String actionPath, List<String> additionalArgs) {
+        return runOnline(modelPath, dataPath, actionPath, "OFF", additionalArgs);
+    }
+
+    public String runOnline(String modelPath, String dataPath, String actionPath, String loggingLevel, List<String> additionalArgs) {
+        List<String> serverArgs = new ArrayList<String>();
+        List<String> clientArgs = new ArrayList<String>();
+        File actionFile = new File(actionPath);
+
+        // Set the server command line arguments.
+        serverArgs.add("--" + CommandLineLoader.OPERATION_INFER_LONG);
+        serverArgs.add("SGDOnlineInference");
+
+        serverArgs.add("--" + CommandLineLoader.OPTION_MODEL_LONG);
+        serverArgs.add(modelPath);
+
+        serverArgs.add("--" + CommandLineLoader.OPTION_DATA_LONG);
+        serverArgs.add(dataPath);
+
+        serverArgs.add("--" + CommandLineLoader.OPTION_OUTPUT_DIR_LONG);
+        serverArgs.add(outDir);
+
+        serverArgs.add("-" + CommandLineLoader.OPTION_PROPERTIES);
+        serverArgs.add("log4j.threshold=" + loggingLevel);
+
+        // Start the server.
+        Runnable serverTask = new Runnable() {
+            @Override
+            public void run() {
+                Launcher.main(serverArgs.toArray(new String[0]), true);
+            }
+        };
+        Thread serverThread = new Thread(serverTask);
```

Comment:
Consider just creating the serverTask anonymous class in this constructor call.
This is purely an optional style.
e.g.:
```
Thread serverThread = new Thread(new Runnable() {
    ...
});
```